File: 1208. Get Equal Substrings Within Budget.cpp
class Solution {
public:
    int equalSubstring(string s, string t, int cost) {
        int n = s.size();
        vector<int> a(n);
        for (int i=0; i<n; ++i)
            a[i] = abs(s[i]-t[i]);
        int ans = 0;
        for (int l=0,r=0,x=0; r<n; ++r) {
            x += a[r];
            while (x > cost)
                x -= a[l++];
            ans = max(ans, r-l+1);
        }
        return ans;
    }
};

File: 1208. Get Equal Substrings Within Budget.py
class Solution:
    def equalSubstring(self, s: str, t: str, cost: int) -> int:
        a = [abs(ord(x)-ord(y)) for x,y in zip(s, t)]
        l = x = ans = 0 
        for r in range(len(s)):
            x += a[r]
            while x > cost:
                x, l = x-a[l], l+1
            ans = max(ans, r-l+1)
        return ans